patches/components-permissions-request_type.cc.patch

```diff
@@ -1,16 +1,24 @@
 diff --git a/components/permissions/request_type.cc b/components/permissions/request_type.cc
-index 7e70c3b195ed2daa118dab140e6323e94b81e8d8..4a0e4b4673a05c99a11cbbd06795817e53574229 100644
+index 7e70c3b195ed2daa118dab140e6323e94b81e8d8..1d93e20a14e55134551054a63d10592edc50cb2f 100644
 --- a/components/permissions/request_type.cc
 +++ b/components/permissions/request_type.cc
-@@ -105,6 +105,7 @@ const gfx::VectorIcon& GetIconIdDesktop(RequestType type) {
+@@ -54,6 +54,7 @@ int GetIconIdAndroid(RequestType type) {
+       return IDR_ANDROID_INFOBAR_PROTECTED_MEDIA_IDENTIFIER;
+     case RequestType::kStorageAccess:
+       return IDR_ANDROID_INFOBAR_PERMISSION_COOKIE;
++    BRAVE_GET_ICON_ID_ANDROID
```

Comment:
can't we just override `GetIconId` or is this a build time error?

Reply:
or better yet, use `GetOverrideIconId` in the `PermissionsClient` subclass if this is not a build time issue

Reply:
please update `BRAVE_GET_ICON_ID_DESKTOP` as well while we're making these changes

Reply:
it was a build time error for unhandling switch cases

Reply:
looks like it can be fixed with a redefine of IDR_ANDROID_INFOBAR_PERMISSION_COOKIE or some other IDR constant.
just store the defined value in a new ```constexpr auto``` variable, then use it instead of the constant in a new ```#define``` and then follow with our custom ```case```s.

Reply:
Fixed in https://github.com/brave/brave-core/pull/9443/commits/63bceae1e84192ea16a32bd815f4fe5674f8f6b3